cmd/prometheus/main.go

```diff
@@ -227,13 +227,15 @@ func reloadConfig(filename string, rls ...Reloadable) (err error) {
 		return fmt.Errorf("couldn't load configuration (-config.file=%s): %v", filename, err)
 	}
 
-	// Apply all configs and return the first error if there were any.
+	failed := false
 	for _, rl := range rls {
-		if err != nil {
-			err = rl.ApplyConfig(conf)
-		} else {
-			rl.ApplyConfig(conf)
+		if err := rl.ApplyConfig(conf); err != nil {
+			log.Error("Failed to apply configuration: ", err)
```

Comment:
That exit differentiation already happens. The question is whether we should log a different message *here* in this function depending on whether it's the first or a subsequent config load. This is also already differentiated in log messages at the caller, just not in this function directly.